### PR TITLE
Let plugin get the key as parameter

### PR DESCRIPTION
**Pull request by oyving**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/AbstractVespaMojo.java [hunk 1]
```diff
@@ -67,8 +70,15 @@ protected void setup() {
         instance = firstNonBlank(instance, project.getProperties().getProperty("instance", "default"));
         id = ApplicationId.from(tenant, application, instance);
 
+        if (privateKey == null || privateKey.isEmpty()) {
+            if (privateKeyFile == null || privateKeyFile.isEmpty()) {
+                throw new IllegalArgumentException("Missing 'privateKey' or 'privateKeyFile' properties.");
+            }
+            privateKey = Paths.get(privateKeyFile);
```
Comment (freva) [inline vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/AbstractVespaMojo.java]: This assigns a `Path` to a `String`?

File: vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/AbstractVespaMojo.java [hunk 2]
```diff
@@ -67,8 +70,15 @@ protected void setup() {
         instance = firstNonBlank(instance, project.getProperties().getProperty("instance", "default"));
         id = ApplicationId.from(tenant, application, instance);
 
+        if (privateKey == null || privateKey.isEmpty()) {
+            if (privateKeyFile == null || privateKeyFile.isEmpty()) {
+                throw new IllegalArgumentException("Missing 'privateKey' or 'privateKeyFile' properties.");
```
Comment (jonmv) [inline vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/AbstractVespaMojo.java]:
```suggestion
                throw new IllegalArgumentException("One of the properties 'privateKey' or 'privateKeyFile' is required.");
```

**Review by jonmv — COMMENTED** [review]
Is this required? Environment variables and properties have a tendency to be printed in logs. 
You'd also have to change the `ai.vespa.hosted.api.Properties` class and its users to reflect the key file no longer being mandatory.

**Comment by jonmv** [comment]
Could you support this in `Properties` as well, so a `-DprivateKey=<key>` affects that in the same way?

**Comment by jonmv** [comment]
LGTM otherwise!

**Pull request merged**